src/main/scala/com/cloudera/sparkts/ARIMA.scala

```diff
@@ -0,0 +1,566 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for
+ * the specific language governing permissions and limitations under the
+ * License.
+ */
+
+package com.cloudera.sparkts
+
+import breeze.linalg._
+
+import org.apache.commons.math3.analysis.{MultivariateVectorFunction, MultivariateFunction}
+import org.apache.commons.math3.optim.nonlinear.scalar.gradient.NonLinearConjugateGradientOptimizer
+import org.apache.commons.math3.optim._
+import org.apache.commons.math3.optim.nonlinear.scalar.{ObjectiveFunctionGradient, GoalType, ObjectiveFunction}
+import org.apache.commons.math3.optim.nonlinear.scalar.noderiv.BOBYQAOptimizer
+import org.apache.commons.math3.random.RandomGenerator
+import org.apache.commons.math3.stat.regression.OLSMultipleLinearRegression
+
+
+object ARIMA {
+  /**
+   * Given a time series, fit a non-seasonal ARIMA model of order (p, d, q), where p represents
+   * the autoregression terms, d represents the order of differencing, and q moving average error
+   * terms. If includeIntercept is true, the model is fitted with a mean. The user can specify
+   * the method used to fit the parameters. The current implementation only supports "css",
+   * meaning conditional sum of squares. In order to select the appropriate order of the model,
+   * users are advised to visualize inspect ACF and PACF plots. Finally, current implementation
+   * does not verify if the parameters fitted are stationary/invertible. Given this, it is suggested
+   * that users inspect the resulting model. However, if parameters are not sane, it is likely
+   * that the order of the model fit is inappropriate.
+   * We follow the specfication outlined in
+   * https://www.otexts.org/fpp/8/5
+   */
+  def fitModel(
+      order:(Int, Int, Int),
+      ts: Vector[Double],
+      includeIntercept: Boolean = true,
+      method: String = "css-BOBYQA"): ARIMAModel = {
+    val (p, d, q) = order
+    val diffedTs = differences(ts, d).toArray.drop(d)
+
+    if (p > 0 && q == 0) {
+      val arModel = Autoregression.fitModel(new DenseVector(diffedTs), p)
+      return new ARIMAModel(order, Array(arModel.c) ++ arModel.coefficients, includeIntercept)
+    }
+
+    // Initial parameter guesses
+    val initParams = HannanRisannenInit(diffedTs, order, includeIntercept)
+
+    // TODO: Enforce stationarity and invertibility for AR and MA terms
+    method match {
+      case "css-BOBYQA" => {
+        val params = fitWithCSSBOBYQA(diffedTs, order, includeIntercept, initParams)
+        new ARIMAModel(order, params, includeIntercept)
+      }
+      case "css-CGD" => {
+        val params = fitWithCSSCGD(diffedTs, order, includeIntercept, initParams)
+        new ARIMAModel(order, params, includeIntercept)
+      }
+      case "ml" => throw new UnsupportedOperationException()
```

Comment:
Should throw an UnsupportedOperationException on any input that's not one of the two css methods